lib/uplinkc/object.go

```diff
@@ -252,11 +270,17 @@ func download_read(downloader C.DownloaderRef, bytes *C.uint8_t, length C.size_t
 		return C.size_t(0)
 	}
 
+	select {
+	case <-download.ctx.Done():
+		return C.size_t(0)
```

Comment:
should you set `cErr`? You could use `download.ctx.Err()` perhaps

Reply:
Haha yes, that's perfect!